### PR TITLE
[REEF-1267] Implement IActiveContext.SubmitContextAndService

### PR DESCRIPTION
**Pull request by afchung**
This addressed the issue by
  * Implements SubmitContextAndService on ActiveContext.
  * Adds a LocalRuntime test to make sure that the SubmitContextAndService call works.

JIRA:
  [REEF-1267](https://issues.apache.org/jira/browse/REEF-1267)

### CODE REVIEW

**Comment by jwang98052** [comment]
I will review it

File: lang/cs/Org.Apache.REEF.Tests/Functional/Bridge/TestContextStack.cs [hunk 1]
```diff
@@ -76,34 +96,101 @@ public IConfiguration DriverConfigurations()
             return TangFactory.GetTang().NewConfigurationBuilder(helloDriverConfiguration).Build();
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Tests/Functional/Bridge/TestContextStack.cs]: You can build in the previos line and return directly.

File: lang/cs/Org.Apache.REEF.Tests/Functional/Bridge/TestContextStack.cs [hunk 2]
```diff
@@ -203,10 +362,42 @@ public byte[] Call(byte[] memento)
             }
         }
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Tests/Functional/Bridge/TestContextStack.cs]: In the CloseContext handler, the close for the first context will be not received. You can verify then remove it from the test.

**Comment by jwang98052** [comment]
I completed the pass. Looks good with a few minor comments.

**Comment by afchung** [comment]
@jwang98052 I've addressed your comments, please have another look. Thanks!

**Comment by jwang98052** [comment]
I will test and nerge

**Pull request closed without merging**